core/src/test/scala/integration/kafka/admin/TopicCommandIntegrationTest.scala

```diff
@@ -732,7 +733,18 @@ class TopicCommandIntegrationTest extends KafkaServerTestHarness with Logging wi
     try {
       killBroker(0)
       val aliveServers = brokers.filterNot(_.config.brokerId == 0)
-      TestUtils.waitForPartitionMetadata(aliveServers, underMinIsrTopic, 0)
+
+      if (isKRaftTest()) {
+        TestUtils.ensureConsistentKRaftMetadata(aliveServers, controllerServer, "Timeout waiting for topic configs propagating to brokers")
+      } else {
+        TestUtils.waitUntilTrue(
+          () => aliveServers.forall(
+            broker =>
+              broker.metadataCache.getPartitionInfo(underMinIsrTopic, 0).get.isr().size() < 6 &&
+                broker.metadataCache.getPartitionInfo(offlineTopic, 0).get.leader() == MetadataResponse.NO_LEADER_ID),
```

Comment:
Why we need the second condition `leader() == MetadataResponse.NO_LEADER_ID)` as well? Could the new leader be elected after killing broker-0?

Reply:
That was suggested by @dengziming  here => https://github.com/apache/kafka/pull/12104#discussion_r865512959 . I haven't got the chance to do the analysis of why that's happening though

Reply:
This `offlineTopic` is used to test the logic of offline partitions, we set its replica assignment to `new NewTopic(offlineTopic, Collections.singletonMap(0, Collections.singletonList(0))` manually, which means it only has one replica, and will have no leader after we kill broker 0.

Reply:
Thanks for the explanation @dengziming !